File: src/main/java/dev/ainterr/minecraft/blockshuffle/gamemodes/RaceGameMode.java
package dev.ainterr.minecraft.blockshuffle.gamemodes;

import dev.ainterr.minecraft.blockshuffle.PlayerList;
import org.bukkit.Material;
import org.bukkit.entity.Player;

public class RaceGameMode implements GameMode {
    private Material block;

    @Override
    public void onRoundStart() {
        this.block = null;
    }

    public void assignBlock(PlayerList players, Player player) {
        if (this.block == null) {
            players.newBlock(player, null);
            this.block = players.getBlockMaterial(player);
        } else {
            players.newBlock(player, this.block);
        }
    }

    public boolean isRoundOver(PlayerList players) {
        return true;
    }
}

File: src/main/java/dev/ainterr/minecraft/blockshuffle/gamemodes/RaceGameMode.kt
package dev.ainterr.minecraft.blockshuffle.gamemodes

import dev.ainterr.minecraft.blockshuffle.PlayerList
import org.bukkit.Material
import org.bukkit.entity.Player

class RaceGameMode : AbstractGameMode() {
    private var block: Material? = null

    override fun onRoundStart() {
        block = null
    }

    override fun assignBlock(players: PlayerList, player: Player?) {
        if (block == null) {
            players.newBlock(player, null)
            block = players.getBlockMaterial(player)
        } else {
            players.newBlock(player, block)
        }
    }

    override fun isRoundOver(players: PlayerList): Boolean {
        return true
    }
}